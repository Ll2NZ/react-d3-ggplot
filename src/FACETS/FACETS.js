import React from "react";
import PropTypes from "prop-types";
import { Fragment } from "react";
import { GGPLOT } from "../GGPLOT/";

function FACETS(props) {
  return (
    <Fragment>
      {props.data.map((dataset, index) => (
        <GGPLOT
          {...props}
          key={index}
          data={dataset} // override old data with only one dataset
        >
          {props.children}
        </GGPLOT>
      ))}
    </Fragment>
  );
}

FACETS.propTypes = {
  data: PropTypes.array,
  children: PropTypes.any
};

export default FACETS;
