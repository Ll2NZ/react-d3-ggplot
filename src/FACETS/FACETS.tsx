import React, { ReactChild } from "react";
import { Fragment } from "react";
import { GGPLOT } from "../GGPLOT/";
import { IContext } from "../_context";

interface IProps extends IContext {
  children: ReactChild;
}

function FACETS(props: IProps) {
  return (
    <Fragment>
      {props.data.map((dataset, index) => (
        <GGPLOT
          data={dataset} // override old data with only one dataset
          aes={props.aes}
          dimensions={props.dimensions}
          key={index}
        >
          {props.children}
        </GGPLOT>
      ))}
    </Fragment>
  );
}

export default FACETS;
