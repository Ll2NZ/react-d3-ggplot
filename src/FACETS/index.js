export { default as FACETS } from "./FACETS.js";
