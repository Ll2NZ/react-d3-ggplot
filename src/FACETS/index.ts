export { default as FACETS } from "./FACETS";
