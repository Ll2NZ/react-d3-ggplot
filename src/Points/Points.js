import React           from "react";
import { Component }   from "react";
import PropTypes       from "prop-types";
import { scaleFinder } from "../Utilities/";
import { select }      from "d3-selection";

class Points extends Component {
    static defaultProps = {
         dimensions:
        {
            width: window.innerWidth*0.9,
            height: window.innerHeight*0.9,
            padding: 50
        },
        className: null,
        color: "orange",
        radius: 1
    }

    static propTypes = {
        data: PropTypes.array.isRequired,
        dimensions: PropTypes.shape({
            width: PropTypes.number,
            height: PropTypes.number,
            padding: PropTypes.number
        }),
        scaleTypes: PropTypes.array,
        className: PropTypes.string,
        aes: PropTypes.array,
        color: PropTypes.string,
        radius: PropTypes.number
    }

    getXScale(){
        // get data and x and y value pair
        const { data, aes } = this.props;

        // convert data to a scale to be used for our x-values
        const xValues  = data.map(item => item[aes[0]]);
        const scaleObj = new scaleFinder(xValues);

        return scaleObj;
    }

    getXScaleType(){
        const { dimensions, scaleTypes } = this.props;
        const scaleObj                   = this.getXScale();
        let xScale;

        // find appropiate scale
        if(scaleTypes[0] === "linear")
            xScale = scaleObj.getLinearScale().nice();

        if(scaleTypes[0] === "time")
            xScale = scaleObj.getTimeScale().nice();

        if(scaleTypes[0] === "ordinal")
            xScale = scaleObj.getOrdinalScale(0.5); // .nice() method not available

        // set scale range
        xScale.range([dimensions.padding, dimensions.width - dimensions.padding]);

        return xScale;
    }

    getYScale(){
       // get y-values
        const { data, aes } = this.props;
        const yValues       = data.map(item => item[aes[1]]);

        // create xScale
        const scaleObj  = new scaleFinder(yValues);

        return scaleObj;
    }

    getYScaleType(){
        const { scaleTypes, dimensions } = this.props;
        const scaleObj                   = this.getYScale();
        let yScale;

        if(scaleTypes[1] === "linear")
            yScale = scaleObj.getLinearScale().nice();

        if(scaleTypes[1] === "time")
            yScale = scaleObj.getTimeScale().nice();

        if(scaleTypes[1] === "ordinal")
            yScale = scaleObj.getOrdinalScale(0.5); // .nice() method not available

        // set scale range
        yScale.range([dimensions.height - dimensions.padding, dimensions.padding]);

        return yScale;
    }

    appendCircles(){
        // scales
        const xScale            = this.getXScaleType();
        const yScale            = this.getYScaleType();

        // all props needed for points
        const { color, radius } = this.props;
        const { data, aes }     = this.props;

        // clear graph for next set of data points
        if(this.node.children.length > 0)
            select(this.node).selectAll("circle").remove();

        // append new points
        select(this.node)
            .selectAll("circle")
            .data(data)
            .enter()
            .append("circle")
            .attr("cx", d => xScale(d[aes[0]]))
            .attr("cy", d => yScale(d[aes[1]]))
            .attr("r", radius)
            .attr("fill", color);
    }

    render(){
        return(
            <g
                ref={ node => this.node = node }
                className={ this.props.className }
            >
            </g>
        );
    }

    componentDidMount(){
        this.appendCircles();
    }

    componentDidUpdate(){
        this.appendCircles();
    }
}

export default Points;
