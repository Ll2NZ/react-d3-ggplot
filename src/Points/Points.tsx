import React from "react";
import { useContext } from "react";
import { ChartContext } from "../_context/";
import { useXScale } from "../_hooks/";
import { useYScale } from "../_hooks/";

interface IProps {
  r?: number;
  fill?: string;
}

Points.displayName = "Points";

function Points(props: IProps) {
  const context = useContext(ChartContext);
  const xScale: any = useXScale();
  const yScale: any = useYScale();

  // will remove later when testing componet
  if (!xScale || !yScale) {
    return <h1>whoops</h1>;
  }

  return (
    <g>
      {/*
        For the key, we're not supposed to used index because we can't take
        advantage of react's optimization capabilities when it comes to re-rendering.
        Will use key until we can sort this detail out.
      */}
      {context.data.map((datum, index) => (
        <circle
          key={index}
          cx={xScale(datum[context.aes[0]])}
          cy={yScale(datum[context.aes[1]])}
          r={!props.r ? 3 : props.r}
          fill={!props.fill ? "orange" : props.fill}
        />
      ))}
    </g>
  );
}

export default Points;
