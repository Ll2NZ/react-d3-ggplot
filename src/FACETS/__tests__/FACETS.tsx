import * as React from "react";
import { render, cleanup } from "react-testing-library";
import FACETS from "../FACETS";
import { Points } from "../../Points/";

// when using facets, we need to have an array of arrays containing objects.
// each entry in the array represents a graph
const DataForFacets = [
  [{ year: 1999, x: 0, y: 0 }, { year: 1999, x: 10, y: 10 }],
  [{ year: 2005, x: 0, y: 0 }, { year: 2005, x: 28, y: 15 }]
];

const validState = {
  data: DataForFacets,
  aes: ["x", "y"],
  dimensions: { width: 500, height: 400, padding: 50 }
};

test("Should render two scatterplots charts", () => {
  const node = render(
    <FACETS {...validState}>
      <Points />
    </FACETS>
  );
});

test("The underlying markup should not change as we change our logic in the code base unless we change the markup ourselves", () => {
  const { container } = render(
    <FACETS {...validState}>
      <Points />
    </FACETS>
  );

  expect(container).toMatchSnapshot();
});
