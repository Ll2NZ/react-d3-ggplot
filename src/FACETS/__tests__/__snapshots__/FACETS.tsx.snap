// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The underlying markup should not change as we change our logic in the code base unless we change the markup ourselves 1`] = `
<div>
  <svg
    height="400"
    width="500"
  >
    <rect
      fill="#f1f1f1"
      height="300"
      transform="translate(50, 50)"
      width="400"
    />
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
      transform="translate(0, 50)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(50.5,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(130.5,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(210.5,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(290.5,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(370.5,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(450.5,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
    </g>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="start"
      transform="translate(50, 0)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(0,350.5)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,290.5)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,230.5)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,170.5)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,110.5)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,50.5)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
    </g>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
      transform="translate(0, 350)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(50.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(90.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          1
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(130.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          2
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(170.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          3
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(210.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          4
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(250.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          5
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(290.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          6
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(330.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          7
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(370.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          8
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(410.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          9
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(450.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          10
        </text>
      </g>
    </g>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="end"
      transform="translate(50, 0)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(0,350.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,320.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          1
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,290.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          2
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,260.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          3
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,230.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          4
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,200.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          5
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,170.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          6
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,140.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          7
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,110.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          8
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,80.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          9
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,50.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          10
        </text>
      </g>
    </g>
    <g>
      <circle
        cx="50"
        cy="350"
        fill="orange"
        r="3"
      />
      <circle
        cx="450"
        cy="50"
        fill="orange"
        r="3"
      />
    </g>
  </svg>
  <svg
    height="400"
    width="500"
  >
    <rect
      fill="#f1f1f1"
      height="300"
      transform="translate(50, 50)"
      width="400"
    />
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
      transform="translate(0, 50)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(50.5,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(121.92857142857143,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(193.35714285714286,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(264.7857142857143,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(336.2142857142857,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(407.64285714285717,0)"
      >
        <line
          opacity="0.7"
          stroke="white"
          y2="300"
        />
      </g>
    </g>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="start"
      transform="translate(50, 0)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(0,350.5)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,256.75)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,163)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,69.25)"
      >
        <line
          opacity="0.7"
          stroke="white"
          x2="400"
        />
      </g>
    </g>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="middle"
      transform="translate(0, 350)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(50.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(79.07142857142857,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          2
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(107.64285714285714,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          4
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(136.21428571428572,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          6
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(164.78571428571428,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          8
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(193.35714285714286,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          10
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(221.92857142857142,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          12
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(250.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          14
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(279.07142857142856,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          16
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(307.64285714285717,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          18
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(336.2142857142857,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          20
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(364.7857142857143,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          22
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(393.35714285714283,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          24
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(421.92857142857144,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          26
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(450.5,0)"
      >
        <line
          stroke="currentColor"
          y2="6"
        />
        <text
          dy="0.71em"
          fill="currentColor"
          y="9"
        >
          28
        </text>
      </g>
    </g>
    <g
      fill="none"
      font-family="sans-serif"
      font-size="10"
      text-anchor="end"
      transform="translate(50, 0)"
    >
      <g
        class="tick"
        opacity="1"
        transform="translate(0,350.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          0
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,313)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          2
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,275.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          4
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,238)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          6
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,200.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          8
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,163)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          10
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,125.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          12
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,88)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          14
        </text>
      </g>
      <g
        class="tick"
        opacity="1"
        transform="translate(0,50.5)"
      >
        <line
          stroke="currentColor"
          x2="-6"
        />
        <text
          dy="0.32em"
          fill="currentColor"
          x="-9"
        >
          16
        </text>
      </g>
    </g>
    <g>
      <circle
        cx="50"
        cy="350"
        fill="orange"
        r="3"
      />
      <circle
        cx="450"
        cy="68.75"
        fill="orange"
        r="3"
      />
    </g>
  </svg>
</div>
`;
